src/jvm/cascading/kryo/KryoSerializer.java

```
@@ -4,65 +4,50 @@
-        // are small (or large), no reallocation occurs.
-        if (prevPosition > KryoSerialization.SWITCH_LIMIT && currentPosition <= KryoSerialization.SWITCH_LIMIT)
-            output.setBuffer(new byte[KryoSerialization.OUTPUT_BUFFER_SIZE], KryoSerialization.MAX_OUTPUT_BUFFER_SIZE);
-
-        prevPosition = currentPosition;
-        output.clear();
-    }
-
-    public void serialize(Object o) throws IOException {
-        kryo.writeObject(output, o);
-        byte[] bytes = output.toBytes();
-
-        outputStream.writeInt(bytes.length);
-        outputStream.write(bytes);
-
-        tidyBuffer();
-    }
-
-    // TODO: Bump the kryo version, add a kryo.reset();
-    public void close() throws IOException {
-        kryo = null;
-
-        try {
-            if( outputStream != null ) {
-                outputStream.close();
-            }
-        } finally {
-            outputStream = null;
-        }
+  private DataOutputStream os;
+  private KryoSerialization ks;
+  private Kryo kryo;
+
+  public KryoSerializer(KryoSerialization kryoSerialization) {
+    ks = kryoSerialization;
+  }
+
+  public void open(OutputStream out) throws IOException {
+    if(out instanceof DataOutputStream)
+      os = (DataOutputStream)out;
+    else
+      os = new DataOutputStream(out);
+    kryo = ks.populatedKryo();
+  }
+
+  public void serialize(Object o) throws IOException {
+    // Write output to temorary buffer.
+    ByteArrayOutputStream bho = new ByteArrayOutputStream();
+    Output ko = new Output(bho);
+    kryo.writeObject(ko, o);
+    ko.flush();
+    // Copy from buffer to output stream.
+    os.writeInt(bho.size());
+    os.write(bho.toByteArray(), 0, bho.size());
+    os.flush();
+  }
+
+  // TODO: Bump the kryo version, add a kryo.reset();
+  public void close() throws IOException {
+    try {
+      if(os != null)
+        os.close();
+    } catch(Exception e) {
+      e.printStackTrace();
```

Are you intentionally swallowing the exception here instead of re-throwing it? If so, why?